test/upgrade/prober/wathola/sender/services.go

```diff
@@ -54,7 +63,11 @@ type sender struct {
 
 func (s *sender) SendContinually() {
 	var shutdownCh = make(chan struct{})
-	defer s.sendFinished()
+	defer func() {
+		s.sendFinished()
+		// Give time to send tracing information.
+		time.Sleep(5 * time.Second)
```

Comment:
I don't like this static wait, as it may introduce failures regarding grace period at later time.

It should be possible to know if tracing is already sent.

Reply:
I don't think there's an easy way to do it. Except for querying back the Zipkin endpoint if the data was stored there (but I wouldn't really like to do this).
There's an open issue in opencensus https://github.com/census-instrumentation/opencensus-go/issues/862
Anyway, the Reporter is created with a default batch interval 1 second. So, it should be enough to wait just 1 second because the data is flushed every 1 second.
Would that be alright? We're already thinking about not shutting down the Sender but using a different way to give it a signal to send the Finished event. So, having it implemented should then relax the need for the 1-second wait time at the end. 
And I think waiting 1 second at the end is not too bad - it's a workaround for the missing feature in opencensus.

Reply:
Right. I think Knative should move to OpenTelemetry as soon as possible. Maybe it will give us greater capabilities as well.

Reply:
Yeah. There's an open issue for moving to OpenTelemetry. It's been there for a while...

Reply:
I'm wondering if this might help: https://github.com/census-instrumentation/opencensus-go/issues/862#issuecomment-476978346?

Reply:
I don't know. It's a different API for exporting "metrics" which we don't need now. And it looks like it can only export those metrics via ReadAndExport. Not the traces :-/ But I'm not 100% sure.